torch/distributed/optim/functional_sgd.py

```diff
@@ -22,7 +22,8 @@ def __init__(
         momentum: float = 0.0,
         dampening: float = 0.0,
         weight_decay: float = 0.0,
-        nesterov: bool = False
+        nesterov: bool = False,
+        allow_empty_param_list: bool = False
```

Comment:
Curious, should we be concerned with TorchScript forward compatibility issues due to changing the number of function arguments? If not, is that because `_FunctionalSGD` is private, so its API is permitted to change?

I also want to mention that I agree with the motivation of this `allow_empty_param_list` argument. I similarly found the need for it when working on overlapping DDP with ZeRO.

Reply:
Regarding FC, indeed a TS interpreter built without this diff would not recognize the new argument, but I don't think this is being used in any such scenarios and _FunctionalSGD is also private.